src/main/java/com/palantir/gradle/versions/VersionsPropsPlugin.java

```diff
@@ -158,9 +158,9 @@ private static void configureDirectDependencyInjection(Configuration conf, Versi
                 versionsProps
                         .getRecommendedVersion(moduleDependency.getModule())
                         .ifPresent(version -> moduleDependency.version(constraint -> {
-                            log.debug("Found direct dependency without version: {} -> {}, preferring: {}",
+                            log.info("Found direct dependency without version: {} -> {}, requiring: {}",
                                     conf, moduleDependency, version);
-                            constraint.prefer(version);
+                            constraint.require(version);
```

Comment:
This also just seems more correct according to:

https://docs.gradle.org/current/userguide/declaring_dependencies.html#sub:declaring_dependency_rich_version